pkg/controller/v1alpha1/localmodelnode/controller_test.go

```diff
@@ -214,20 +278,86 @@ var _ = Describe("CachedModel controller", func() {
 			Expect(k8sClient.Create(context.TODO(), configMap)).NotTo(HaveOccurred())
 			defer k8sClient.Delete(context.TODO(), configMap)
 
-			// Mock readDir to return a fake model folder
-			readDir = func(_ string) ([]fs.DirEntry, error) {
-				return []fs.DirEntry{
-					&MockFileInfo{name: modelName, isDir: true},
-				}, nil
+			clusterStorageContainer := &v1alpha1.ClusterStorageContainer{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+				},
+				Spec: clusterStorageContainerSpec,
 			}
+			Expect(k8sClient.Create(ctx, clusterStorageContainer)).Should(Succeed())
+			defer k8sClient.Delete(ctx, clusterStorageContainer)
 
-			removeAllCalled := false
-			var pathRemoved string
-			removeAll = func(path string) error {
-				pathRemoved = path
-				removeAllCalled = true
-				return nil
+			nodeGroup := &v1alpha1.LocalModelNodeGroup{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "gpu",
+				},
+				Spec: localModelNodeGroupSpec,
 			}
+			Expect(k8sClient.Create(ctx, nodeGroup)).Should(Succeed())
+			defer k8sClient.Delete(ctx, nodeGroup)
+
+			nodeName = "worker2"
+			localModelNode := &v1alpha1.LocalModelNode{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: nodeName,
+				},
+				Spec: localModelNodeSpec,
+			}
+			Expect(k8sClient.Create(ctx, localModelNode)).Should(Succeed())
+			// defer k8sClient.Delete(ctx, localModelNode)
+
+			// Wait for the download job to be created
+			job := &batchv1.Job{}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: modelName + "-" + nodeName, Namespace: modelCacheNamespace}, job)
+				return err == nil
+			}, timeout, interval).Should(BeTrue(), "Download job should be created")
+
+			// Now let's update the job status to be successful
+			fsMock.mockDir(modelsRootFolder, []fs.DirEntry{
+				&MockFileInfo{name: modelName, isDir: true},
+			})
+
+			job.Status.Succeeded = 1
+			Expect(k8sClient.Status().Update(ctx, job)).Should(Succeed())
+
+			// Delete the model folder
+			fsMock.mockDir(modelsRootFolder, []fs.DirEntry{})
+
+			// Manually trigger reconcillation
+			patch := client.MergeFrom(localModelNode.DeepCopy())
+			localModelNode.Annotations = map[string]string{"foo": "bar"}
+			Expect(k8sClient.Patch(ctx, localModelNode, patch)).Should(Succeed())
+
+			newJob := &batchv1.Job{}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: modelName + "-" + nodeName, Namespace: modelCacheNamespace}, newJob)
+				// Todo: Figure out why we have to manually remove finalizers in the test. It works in kind clusters.
```

Comment:
In this test the finalizer seems never removed, but it works in my kind cluster.

Reply:
basically the job cannot be cleaned up in the test for some reason